gateway/res_handler_go_plugin.go

```diff
@@ -0,0 +1,94 @@
+package gateway
+
+import (
+	"fmt"
+	"github.com/TykTechnologies/tyk/apidef"
+	"github.com/TykTechnologies/tyk/goplugin"
+	"github.com/TykTechnologies/tyk/user"
+	"github.com/sirupsen/logrus"
+	"net/http"
+	"time"
+)
+
+type ResponseGoPluginMiddleware struct {
+	Path       string // path to .so file
+	SymbolName string // function symbol to look up
+	logger     *logrus.Entry
+	Spec       *APISpec
+	ResHandler func(rw http.ResponseWriter, res *http.Response, req *http.Request, ses *user.SessionState)
```

Comment:
We must have this method signature to satisfy the response handler interface

Reply:
Lets drop it from interface as well. And for compatibility reasons second line of the response function will be like:
`ses = ctx.GetSession(req)`

Reply:
Should we drop the req object too - can we get that from context also?

Reply:
But our request plugins already have different signature already. And if we keep Go response plugins out of sync with request plugins it will also look bad. 
Go plugins have completely different nature, and benefits. I think it worth to have a bit separate contract for them.

Reply:
I'll just add a method to the TykResponseHandler with that sig to handle the go plugin cases

Reply:
Yep, works for me